sdk/test-utils/recorder/test/browser/recorder.spec.ts

```diff
@@ -0,0 +1,311 @@
+    (window as any).__env__.TEST_MODE = "record";
+    (window as any).__env__.PATH = "/to/replace";
+
+    // We can't use Nise's FakeServer since the recorder ends up sending the request through the original XHR anyway.
+    xhrMock.setup();
+    xhrMock.get("/to/replace", {
+      status: 200,
+      body: expectedHttpResponse
+    });
+
+    // Before starting the recorder, we need to make a copy of the original XHR object, so that we can
+    // restore it before doing the recorder do its magic, in order for the HTTP request to hit the xhr-mock instance.
+    const originalXHR = XMLHttpRequest;
+
+    // The recorder outputs files into the console,
+    // so we need to mock the console.log function to capture and test the recorder output.
+    const originalConsoleLog = console.log;
+    const savedConsoleLogParams: any[] = [];
+    console.log = (...params: any) => {
+      savedConsoleLogParams.push(params);
+    };
+
+    // The recorder should start in the beforeEach call.
+    // We have to do this to emulate that.
+    const fakeThis: any = {
+      ...this,
+      currentTest: {
+        file: "test/recorder.browser.spec.ts",
+        // For this test, we don't care what's the content of the recorded function.
+        fn: getNoOpFunction()
+      }
+    };
+
+    const recorder = record(fakeThis, recorderEnvSetup);
+
+    // Restoring the XHR, otherwise we can't test this.
+    window.XMLHttpRequest = originalXHR;
+
+    const response = await helloWorldRequest();
+
+    // This test's request reached the server and received the expected response.
+    expect(response).to.equal(expectedHttpResponse);
+
+    // Cleaning everything before we continue verifying the results.
+    xhrMock.teardown();
+    recorder.stop();
+    console.log = originalConsoleLog;
+
+    // Here we confirm that the recorder generated an expected output on the console.logs.
+    // This output is used to generate the recording files in the filesystem, though here we're only
+    // checking what was that the recorded emitted to the standard output.
+    expect(savedConsoleLogParams[0][0]).to.equal(
+      // The recordings here are empty because we hijacked the XHR.
+      // See the playback test for an example of a properly constructed recording object.
+      // TODO: Find a way to capture the complete output.
+      JSON.stringify({
+        writeFile: true,
+        path:
+          "./recordings/browsers/the_recorders_public_api_on_a_browser/recording_should_record_a_simple_test.json",
+        content: {
+          recordings: [],
+          uniqueTestInfo: {
+            uniqueName: {},
+            newDate: {}
+          },
+          hash: "7c1368cdbc23988b15934bf7655c765b"
+        }
+      })
+    );
+  });
+
+  it("should playback a simple test", async function() {
+    // Setting up the playback mode.
+    // The PATH environment variable is not needed on playback.
+    (window as any).__env__.TEST_MODE = "playback";
```

Comment:
By default TEST_MODE is `playback`, so even this line is not required.

Reply:
ohh! thanks!